.github/workflows/ci.yml

```diff
@@ -1,23 +1,44 @@
 ---
-name: build
+name: test
 on:
   push:
     branches:
       - master
   pull_request:
-    branches:
-      - master
+    types: [opened, synchronize]
+
 jobs:
-  ci:
-    name: >
-      Run checks and tests over ${{matrix.otp_vsn}} and ${{matrix.os}}
-    runs-on: ${{matrix.os}}
-    container:
-      image: erlang:${{matrix.otp_vsn}}
+  test:
+    name: Erlang/OTP ${{matrix.otp}} / rebar3 ${{matrix.rebar3}}
+    runs-on: ubuntu-latest
     strategy:
       matrix:
-        otp_vsn: [19.3, 20.3, 21.3, 22.3, 23.2]
-        os: [ubuntu-latest]
+        otp: ['20.3.8.9', '21.3.8.9', '22.3.4.9', '23.3.1']
```

Comment:
Why not use '20', instead of '20.3.8.9'? The action will pick the latest for you 😄

Reply:
It probably should yeah, I copied what was done in verl and that was just testing a good bit for the most part, but yeah agree to use majors only.